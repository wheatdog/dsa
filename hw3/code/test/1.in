1+2*(8/4) % 5
(+52 * -4) << (1 & 2 | 3) 
(!40 || 2 + ~30) ^ (2 >> 1)
