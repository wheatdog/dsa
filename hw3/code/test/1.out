--- postfix expression transforming ---
encounter 1: push to output
	current output: 1
encounter +: push to the stack directly
	current output: 1
	current stack:  +
encounter 2: push to output
	current output: 1 2
	current stack:  +
encounter *: push to the stack directly
	current output: 1 2
	current stack:  + *
encounter (: push to the stack directly
	current output: 1 2
	current stack:  + * (
encounter 8: push to output
	current output: 1 2 8
	current stack:  + * (
encounter /: push to the stack directly
	current output: 1 2 8
	current stack:  + * ( /
encounter 4: push to output
	current output: 1 2 8 4
	current stack:  + * ( /
encounter ): flush the stack to output until meeting '('
	current output: 1 2 8 4 /
	current stack:  + *
encounter %: stack.top() has greater or the same precdence, after pop something out to output, then push to the stack
	current output: 1 2 8 4 / *
	current stack:  + %
encounter 5: push to output
	current output: 1 2 8 4 / * 5
	current stack:  + %
encounter NOTHING: flush the stack to output
	current output: 1 2 8 4 / * 5 % +
--- postfix expression transforming complete :) ---
Postfix Exp: 1 2 8 4 / * 5 % +
RESULT: 5
--- postfix expression transforming ---
encounter (: push to the stack directly
	current output:
	current stack:  (
encounter U+: push to the stack directly
	current output:
	current stack:  ( +
encounter 52: push to output
	current output: 52
	current stack:  ( +
encounter *: stack.top() has greater or the same precdence, after pop something out to output, then push to the stack
	current output: 52 +
	current stack:  ( *
encounter U-: push to the stack directly
	current output: 52 +
	current stack:  ( * -
encounter 4: push to output
	current output: 52 + 4
	current stack:  ( * -
encounter ): flush the stack to output until meeting '('
	current output: 52 + 4 - *
encounter <<: push to the stack directly
	current output: 52 + 4 - *
	current stack:  <<
encounter (: push to the stack directly
	current output: 52 + 4 - *
	current stack:  << (
encounter 1: push to output
	current output: 52 + 4 - * 1
	current stack:  << (
encounter &: push to the stack directly
	current output: 52 + 4 - * 1
	current stack:  << ( &
encounter 2: push to output
	current output: 52 + 4 - * 1 2
	current stack:  << ( &
encounter |: stack.top() has greater or the same precdence, after pop something out to output, then push to the stack
	current output: 52 + 4 - * 1 2 &
	current stack:  << ( |
encounter 3: push to output
	current output: 52 + 4 - * 1 2 & 3
	current stack:  << ( |
encounter ): flush the stack to output until meeting '('
	current output: 52 + 4 - * 1 2 & 3 |
	current stack:  <<
encounter NOTHING: flush the stack to output
	current output: 52 + 4 - * 1 2 & 3 | <<
--- postfix expression transforming complete :) ---
Postfix Exp: 52 + 4 - * 1 2 & 3 | <<
RESULT: -1664
--- postfix expression transforming ---
encounter (: push to the stack directly
	current output:
	current stack:  (
encounter !: push to the stack directly
	current output:
	current stack:  ( !
encounter 40: push to output
	current output: 40
	current stack:  ( !
encounter ||: stack.top() has greater or the same precdence, after pop something out to output, then push to the stack
	current output: 40 !
	current stack:  ( ||
encounter 2: push to output
	current output: 40 ! 2
	current stack:  ( ||
encounter +: push to the stack directly
	current output: 40 ! 2
	current stack:  ( || +
encounter ~: push to the stack directly
	current output: 40 ! 2
	current stack:  ( || + ~
encounter 30: push to output
	current output: 40 ! 2 30
	current stack:  ( || + ~
encounter ): flush the stack to output until meeting '('
	current output: 40 ! 2 30 ~ + ||
encounter ^: push to the stack directly
	current output: 40 ! 2 30 ~ + ||
	current stack:  ^
encounter (: push to the stack directly
	current output: 40 ! 2 30 ~ + ||
	current stack:  ^ (
encounter 2: push to output
	current output: 40 ! 2 30 ~ + || 2
	current stack:  ^ (
encounter >>: push to the stack directly
	current output: 40 ! 2 30 ~ + || 2
	current stack:  ^ ( >>
encounter 1: push to output
	current output: 40 ! 2 30 ~ + || 2 1
	current stack:  ^ ( >>
encounter ): flush the stack to output until meeting '('
	current output: 40 ! 2 30 ~ + || 2 1 >>
	current stack:  ^
encounter NOTHING: flush the stack to output
	current output: 40 ! 2 30 ~ + || 2 1 >> ^
--- postfix expression transforming complete :) ---
Postfix Exp: 40 ! 2 30 ~ + || 2 1 >> ^
RESULT: 0
